p4-16/spec/P4-16-draft-spec.mdk

```diff
@@ -2246,6 +2269,181 @@ manipulate the elements at the front and back of the stack:
   ```count``` argument must be a positive integer that is a
   compile-time known value. The return type is ```void```.
 
+## Operations on Header Unions { #sec-expr-hu }
+
+A variable declared with a union type is initially invalid:
+
+~Begin P4Example
+header H1 {
+  bit<8> f;
+}  
+
+header H2 {
+  bit<16> g;
+}  
+
+header_union U {
+  H1 h1;
+  H2 h2;
+}
+
+U u; // u invalid
+~End P4Example
+
+This also implies that each of the headers `h1` through `hn` contained
+in it is also invalid. Unlike headers, a union cannot be initialized.
+
+However, we can change the validity of a header union by assigning a
+valid header to one of its elements:
+~Begin P4Example
+header H1
+U u; 
+H1 my_h1 = { 8w0 }; // my_h1 is valid
```

Comment:
I support having a similar example with tuples assigned to headers there. But let's keep this here.